Drag and drop enabled table and Participate Tab

forkitupforkitup opened the pull request:
This PR addresses [issue #29.](https://github.com/codeforsanjose/gov-agenda-notifier/issues/29)
![giphy](https://user-images.githubusercontent.com/66397920/100187167-62112a00-2e9c-11eb-86fd-17645fef5212.gif)
Hello,
This franken-table may not play nice with the background-color of the page. The table dynamically changes its height based on how many rows it has, and because of the order in which React components are mounted, it could stick out beyond the height of the page. I chose to make it as big as the number of rows because then I don't have to use the default CSS overflow property of the library that produces the table.
Another item that needs work is making the drop down box show its options either above or below the currently selected option depending on where there's enough space to show all the options.
Looking forward to your review and suggestions.

forkitupforkitup commented:
Just resolved conflicts with master

aaa12345678910 commented:
Can you provide an example of embedded agenda items?

forkitupforkitup commented:
> Can you provide an example of embedded agenda items?

I haven't started on that yet. I'd like to start that before hack night on Thursday

teohrt commented on frontend/package.json:
```diff
@@ -7,35 +7,38 @@
     "passport": "^0.4.1",
     "passport-google-oauth20": "^2.0.0",
     "passport-jwt": "^4.0.0",
-    "pg": "^8.3.3",
```
Is there a reason for a change in this package version?

forkitupforkitup replied:
Hi Trace, sorry to get back to you so late. I think it's because the  carat at the beginning made it so it would automatically update. I don't think we need postgres in the frontend? I'm not sure.

teohrt commented on backend/graphql_api/lambda/auth/passportSetup.js:
```diff
@@ -1,40 +1,46 @@
-const passport = require('passport');
```
This PR has formatted many files and that raises a red flag. It's not always a bad thing, but in this case the PR is styling many files inappropriately.

The main issue is the removal of semi colons. Semi colons aren't required but they're a good idea to have as their absence can raise a series of unexpected bugs. 

Explicit > implicit. https://github.com/airbnb/javascript/issues/1567

I'm ok with differing style guides between frontend and backend. Sometimes that's appropriate and there should be a discussion about it if necessary. But it looks like your IDE is using linting settings that aren't set within the repo. Check out `backend/.eslintrc.json` for an example.

I would suggest you update your eslint settings in your IDE and reapply your formatting changes, or simply omit the modifications having to do with semi colons in the backend directory.

forkitupforkitup replied:
I'm open to having semicolons and don't mind either way - I just installed prettier and wasn't fully aware of the implications. Let me take a look at Prettier's settings.

teohrt reviewed (changes requested):
There's a minor linting issue, but great PR!

teohrt commented:
This PR has brought linting to the forefront of my attention. We need to standardize so our PRs don't include 50+ file modifications due to linting config mismatching.

forkitupforkitup commented:
Hi all, I rebased and dropped the commits `6cb31eb`, and `174c8de` on my local repo. These were the ones causing linting issues when I tried merging my branch into master. `3a1111a` takes `1aca8ae` and simply removes all prettier configuration, and sticks with the long standing ESLint config (no enforcement of code style or finding problems). I just need something to demo my progress tonight.
